source/pdal/pdalc_pipeline.cpp

```diff
@@ -222,80 +217,122 @@ extern "C"
     {
         Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
 
-        if (ptr && ptr->get())
+        try
         {
-            try
-            {
-                ptr->get()->setLogLevel(level);
-            }
-            catch (const std::exception &e)
-            {
-                printf("Found error while setting log level: %s\n", e.what());
-            }
+            if (level < 0 || level > 8)
+                throw pdal_error("log level must be between 0 and 8!");
+
+            pdal::LogPtr lptr = ptr->manager->log();
+            lptr->setLevel(static_cast<pdal::LogLevel>(level));
+        }
+        catch (const std::exception &e)
+        {
+            printf("Found error while setting log level: %s\n", e.what());
         }
+
     }
 
     int PDALGetPipelineLogLevel(PDALPipelinePtr pipeline)
     {
         Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
-        return (ptr && ptr->get()) ? ptr->get()->getLogLevel() : 0;
+        try
+        {
+            return (ptr)
+                   ? static_cast<int>(
+                       ptr->manager->log()->getLevel()
+                   ) : 0;
+        }
+        catch (const std::exception &e)
+        {
+            printf("Found error while getting log level: %s\n", e.what());
+        }
+        return 0;
     }
 
     int64_t PDALExecutePipeline(PDALPipelinePtr pipeline)
     {
         int64_t result = 0;
         Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
 
-        if (ptr && ptr->get())
+        if (ptr)
         {
             try
             {
-                result = ptr->get()->execute();
+                result = ptr->manager->execute();
+                ptr->m_executed = true;
             }
             catch (const std::exception &e)
             {
                 printf("Found error while executing pipeline: %s", e.what());
             }
         }
-
         return result;
     }
 
+    bool PDALExecutePipelineAsStream(PDALPipelinePtr pipeline)
+    {
+        Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
+
+        if (ptr)
+        {
+            try
+            {
+                PipelineManager::ExecResult exec = ptr->manager->execute(ExecMode::Stream);
+                ptr->m_executed = true;
+                return true;
+            }
+            catch (const std::exception &e)
+            {
+                printf("Found error while executing pipeline: %s", e.what());
+                return false;
+            }
+        }
+        return false;
+    }
+
+    bool PDALPipelineIsStreamable(PDALPipelinePtr pipeline)
+    {
+        Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
```

Comment:
This would be cleaner if you just check the pointer at input:
```
if (! pipeline)
    return false;
Pipeline *ptr = reinterpret_cast<...>(pipeline);
return ptr->manager->pipelineStreamable;
```

Reply:
lgtm